blockchain/blockchain.go

```diff
@@ -500,6 +503,29 @@ func (bc *blockchain) GetActionsFromAddress(addrStr string) ([]hash.Hash256, err
 	return getActionsBySenderAddress(bc.dao.kvstore, hash.BytesToHash160(addr.Bytes()))
 }
 
+// GetActionsFromIndex returns actions from index
+func (bc *blockchain) GetActionHashFromIndex(index uint64) (hash.Hash256, error) {
+	hash := hash.ZeroHash256
+	value, err := bc.dao.kvstore.Get(blockActionBlockMappingNS, indexActionsTipIndexKey)
```

Comment:
I think as long as you can find the has for the index, you could find the action, otherwise return not found. Checking against tipIndexActions is not necessary

Reply:
I'll test without check tipIndexActions